presto-main/src/main/java/io/prestosql/operator/LookupJoinOperator.java

```diff
@@ -677,14 +677,14 @@ private boolean outerJoinCurrentPosition()
             }
 
             boolean consumedInput = false;
-            if (needsInput()) {
-                addInput(element);
+            if (needsInput() && inputPage != null) {
```

Comment:
But isn't `inputPage != null` redundant here ? Bcoz `needsInput()` will be false once `finishing` is set to true which happens in this snippet
```
if (innputPage == null) {
    finish();
}
```

Reply:
It's not redundant. `finish()` might not set `finishing` when spill is in progress. Therefore `needsInput` might be true once the spill is done, but `inputPage` is null